database_management_bot.py

```diff
@@ -0,0 +1,2235 @@
+    The insert_to_db function adds a drug to the database.
+    It takes in an update and context as arguments, checks the user confirnation message,
+    and if it's 'Так, додати до бази даних' then adds it to the database.
+    It also sends a message confirming that it was added successfully.
+    If message text is 'Змінити інформацію' it returns change_info of the conversation.
+    If none of that checks are passed it cancels the adding process
+
+    :param update: Update: Access the message object
+    :param context: CallbackContext: Keep track of the state of a conversation
+    :return: Next state or ConversationHandler.END
+    """
+    user = update.message.from_user
+
+    reply_keyboard = MAIN_REPLY_KEYBOARD
+
+    if update.message.text == 'Так, додати до бази даних':
+        user_id = update.effective_user.id
+
+        context.user_data["DRUG_INFO"]["user_id"] = user_id
+        context.user_data["DRUG_INFO"]["added_on"] = datetime.now().strftime("%d/%m/%Y, %H:%M:%S")
+
+        collection.insert_one(context.user_data["DRUG_INFO"])
+        logger.info("Checked info. Added to DB successfully")
+
+        if "query" in context.user_data:
+            context.user_data["query"].edit_message_text(
+                text="✅ Ви додали інформацію про цей штрих\-код \- "
+                     f"*{context.user_data['DRUG_INFO']['code']}*"
+                     "\n\n*Дякуємо\!*",
+                parse_mode="MarkdownV2",
+            )
+            context.user_data.pop("query", None)
+
+        update.message.reply_text(
+            text='✅ Препарат успішно додано до бази даних',
+            reply_markup=ReplyKeyboardMarkup(
+                reply_keyboard,
+                one_time_keyboard=True,
+                resize_keyboard=True,
+                input_field_placeholder="Оберіть опцію",
+            ),
+        )
+
+        logger.info("Clearing info")
+        context.user_data["DRUG_INFO"].clear()
+        logger.info("Cleared")
+
+    elif update.message.text == 'Змінити інформацію':
+        logger.info("User asked to change info")
+
+        reply_keyboard_change = [['Назва', 'Діюча речовина', 'Опис']]
+        update.message.reply_text(
+            text='Гаразд, яке поле ви хочете змінити?',
+            reply_markup=ReplyKeyboardMarkup(
+                reply_keyboard_change,
+                one_time_keyboard=True,
+                resize_keyboard=True,
+                input_field_placeholder="Оберіть опцію",
+            ),
+        )
+        return CHANGE_INFO
+
+    else:
+        logger.info("User %s canceled adding process", user.first_name)
+
+        update.message.reply_text(
+            text='☑️ Гаразд, додавання скасовано',
+            reply_markup=ReplyKeyboardMarkup(
+                reply_keyboard,
+                one_time_keyboard=True,
+                resize_keyboard=True,
+                input_field_placeholder="Оберіть опцію",
+            ),
+        )
+
+    return ConversationHandler.END
```

Comment:
It's better to clear `DRUG_INFO` also after canceling.
```suggestion
    elif update.message.text == 'Змінити інформацію':
        logger.info("User asked to change info")

        reply_keyboard_change = [['Назва', 'Діюча речовина', 'Опис']]
        update.message.reply_text(
            text='Гаразд, яке поле ви хочете змінити?',
            reply_markup=ReplyKeyboardMarkup(
                reply_keyboard_change,
                one_time_keyboard=True,
                resize_keyboard=True,
                input_field_placeholder="Оберіть опцію",
            ),
        )
        return CHANGE_INFO

    else:
        logger.info("User %s canceled adding process", user.first_name)

        update.message.reply_text(
            text='☑️ Гаразд, додавання скасовано',
            reply_markup=ReplyKeyboardMarkup(
                reply_keyboard,
                one_time_keyboard=True,
                resize_keyboard=True,
                input_field_placeholder="Оберіть опцію",
            ),
        )
    logger.info("Clearing info")
    context.user_data["DRUG_INFO"].clear()
    logger.info("Cleared")

    return ConversationHandler.END
```

Reply:
Clearing `DRUG_INFO` is not required here at this time, as a bot's persistence is planned to be introduced (read more about persistence [here](https://github.com/python-telegram-bot/v13.x-wiki/wiki/Making-your-bot-persistent)), allowing users to continue the adding process after cancellation or moving on to another action.